marker_api/src/sem/item.rs

```diff
@@ -80,57 +75,52 @@ impl<'ast> Visibility<'ast> {
+    /// // Returns `true` since this item is at the root of the crate and the default
+    /// // visibility is therefore `pub(crate)`
     /// fn example_in_root() {}
+    ///
+    /// # fn main() {}
     /// ```
-    pub fn is_pub_crate(&self) -> bool {
-        matches!(self.kind, VisibilityKind::Crate(_))
+    pub fn is_crate_scoped(&self) -> bool {
+        matches!(self.kind, VisibilityKind::Crate(_) | VisibilityKind::DefaultCrate(_))
     }
 
     /// Returns `true` if a visibility is the default visibility, meaning that it wasn't
     /// declared.
     pub fn is_default(&self) -> bool {
-        matches!(self.kind, VisibilityKind::Default(_) | VisibilityKind::DefaultPub)
-    }
-
-    /// Returns the [`ItemId`] of the module where this item is visible in, if the
-    /// visibility is defined to be public inside a specified path.
-    ///
-    /// See [`Visibility::module_id`] to get the `ItemId`, even if the item or
-    /// field uses the default visibility.
-    pub fn pub_with_path_module_id(&self) -> Option<ItemId> {
-        match self.kind {
-            VisibilityKind::Path(id) | VisibilityKind::Crate(id) => Some(id),
-            _ => None,
-        }
+        matches!(
+            self.kind,
+            VisibilityKind::DefaultPub | VisibilityKind::DefaultCrate(_) | VisibilityKind::Default(_)
+        )
     }
 
     /// Returns the [`ItemId`] of the module that this item or field is visible in.
-    /// It will return `None`, if the item is public, as the visibility extends even past
+    /// It will return `None`, if the item is public, as the visibility extends past
     /// the root module of the crate.
     ///
     /// This function also works for items which have the default visibility, of the
     /// module they are declared in.
     ///
     /// ```
+    /// // lib.rs
+    ///
     /// mod scope {
     ///     // Returns `None` since this is even visible outside the current crate
     ///     pub fn turtle() {}
     ///     
     ///     // Returns the `ItemId` of the root module of the crate
     ///     pub(crate) fn shark() {}
     ///
-    ///     // Returns the `ItemId` of the module it is declared in
+    ///     // Returns the `ItemId` of the module it is declared in, in this case `crate::scope`
     ///     fn dolphin() {}
     /// }
     /// ```
-    ///
-    /// Note that this only returns the [`ItemId`] that this item is visible in
-    /// based on the declared visibility. The item might be reexported, which can
-    /// increase the visibility.
-    pub fn module_id(&self) -> Option<ItemId> {
+    pub fn scope(&self) -> Option<ItemId> {
         match self.kind {
-            VisibilityKind::Path(id) | VisibilityKind::Crate(id) | VisibilityKind::Default(id) => Some(id),
+            VisibilityKind::Path(id)
+            | VisibilityKind::Crate(id)
+            | VisibilityKind::DefaultCrate(id)
+            | VisibilityKind::Default(id) => Some(id),
             _ => None,
```

Comment:
`[nit]`
I'd prefer an exhaustive match to make sure all cases are handled if `VisibilityKind` enum is modified